homeassistant/components/enocean/config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Config flows for the ENOcean integration."""
+from enum import Enum
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import CONN_CLASS_ASSUMED
+from homeassistant.const import CONF_DEVICE
+
+from . import EnOceanDongle
+from .const import DOMAIN, ERROR_INVALID_DONGLE_PATH
+
+
+class DongleSetupStates(Enum):
+    """The statuses of the enocean dongle configuration state machine."""
+
+    SELECT_AUTO = 1
+    SELECT_MANUAL = 2
+    VALIDATE = 3
+    CREATE_ENTRY = 4
+
+
+class EnOceanFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the enOcean config flows."""
+
+    VERSION = 1
+    MANUAL_PATH_VALUE = "Custom path"
+    CONNECTION_CLASS = CONN_CLASS_ASSUMED
+
+    def __init__(self):
+        """Initialize the EnOcean config flow."""
+        self.dongle_path = None
+        self.discovery_info = None
+
+    async def async_step_import(self, data=None):
+        """Import a yaml configuration."""
+        return self.async_create_entry(title="EnOcean", data=data)
```

Comment:
We want to check the connection to the dongle before creating the entry for import too.

Reply:
Added the check. If it fails, I return async_abort. It does not seem to display an error in the UI. I also have added a warning log.